### PR TITLE
DT-604 initial commit to wire in actual court hearing creation into API endpoint.

### PR DESCRIPTION
**Pull request by matt-crowson-moj**
Removing stubbed API return value from endpoint and wiring in call to service layer to schedule (create) court hearings and return via API endpoint.

This also fixes a bug I found during testing with the case status being active.

### CODE REVIEW

File: src/main/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImpl.java [hunk 1]
```diff
@@ -1,36 +1,29 @@
...
-                .id(-1L)
-                .date(hearing.getCourtHearingDateTime().toLocalDate())
-                .time(hearing.getCourtHearingDateTime().toLocalTime())
-                .location(Agency.builder()
-                        .agencyId(hearing.getToCourtLocation())
-                        .active(true)
-                        .agencyType("CRT")
-                        .build())
-                .build());
+        return ResponseEntity.status(CREATED).body(courtHearingsService.scheduleHearing(bookingId, hearing));
```
Comment (andymarke) [inline src/main/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImpl.java]: Can you just return a CourtHearing? I guess there is an annotation to say 201 response code?

File: src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest.java [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+
+public class OffenderMovementsResourceImplIntTest extends ResourceTest {
+
+    @Test
+    public void schedules_court_hearing() {
+        final var token = authTokenHelper.getToken(AuthTokenHelper.AuthToken.NORMAL_USER);
+
+        final var request = createHttpEntity(token, Map.of(
+                "courtCaseId", -1,
+                "fromPrisonLocation", "LEI",
```
Comment (andymarke) [inline src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest.java]: Sort on interesting that you need a prison location to create a court hearing, I wonder what happens if they are being held in the court and there are two hearings, would the fromPrisonLocation still be a prison? I guess this models what you have to input to NOMIS

File: src/main/java/net/syscon/elite/service/CourtHearingsService.java [hunk 3]
```diff
@@ -56,6 +57,7 @@ public CourtHearingsService(final OffenderBookingRepository offenderBookingRepos
     }
 
     @Transactional
+    @VerifyBookingAccess
     public CourtHearing scheduleHearing(final Long bookingId, final PrisonToCourtHearing hearing) {
```
Comment (Mjwillis) [inline src/main/java/net/syscon/elite/service/CourtHearingsService.java]: Probably will want the `@HasWriteScope`

Comment (Mjwillis) [reply src/main/java/net/syscon/elite/service/CourtHearingsService.java]: On the resource you'll also want `@ProxyUser`

File: src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest.java (hunk 2 above)
Comment (matt-crowson-moj) [reply src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest.java]: Indeed it is really supporting what NOMIS does at this point in time.  Happy to talk about though.

File: src/main/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImpl.java (hunk 1 above)
Comment (matt-crowson-moj) [reply src/main/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImpl.java]: The challenge in (some respects) is looking for consistency in the elite code base.  It isn't always consistent.

File: src/main/java/net/syscon/elite/service/CourtHearingsService.java (hunk 3 above)
Comment (matt-crowson-moj) [reply src/main/java/net/syscon/elite/service/CourtHearingsService.java]: Done

File: src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+                ErrorResponse.builder()
+                        .status(404)
+                        .userMessage("Resource with id [9999999] not found.")
+                        .developerMessage("Resource with id [9999999] not found.")
+                        .build());
+    }
+
+    @Test
+    public void schedule_court_hearing_fails_when_no_matching_prison() {
+        final var token = authTokenHelper.getToken(AuthTokenHelper.AuthToken.NORMAL_USER);
```
Comment (Mjwillis) [inline src/test/java/net/syscon/elite/api/resource/impl/OffenderMovementsResourceImplIntTest.java]: I’d add a test for a user without permissions or write scope access

**Pull request merged**